src/bbs2gh/Program.cs

```diff
@@ -33,7 +33,9 @@ public static async Task Main(string[] args)
                 .AddSingleton<VersionChecker>()
                 .AddSingleton<HttpDownloadService>()
                 .AddSingleton<IVersionProvider, VersionChecker>(sp => sp.GetRequiredService<VersionChecker>())
+                .AddSingleton<FileSystemProvider>()
                 .AddTransient<ITargetGithubApiFactory>(sp => sp.GetRequiredService<GithubApiFactory>())
+                .AddTransient<BbsArchiveDownloaderFactory>()
```

Comment:
Why isn't this a Singleton?

Reply:
Good question, it doesn't need to be but since all other registrations are Singletons I guess we can make this a Singleton too and refactor other non-Singleton one to be Singletons too.